doctrine.rst

```diff
@@ -412,6 +412,76 @@ Take a look at the previous example in more detail:
+         * @Route("/product", name="product")
+         */
+        public function index(ValidatorInterface $validator)
+        {
+            $product = new Product();
+            $product->setName(null); // The column in database isn't nullable
+            $product->setPrice('1999'); // Type mismatch, an integer is expected
+
+            // ...
+
+            $errors = $validator->validate($product);
+            if (count($errors) > 0) {
+                return new Response((string) $errors, 400);
+            }
+
+            // Will not be reached in this example
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($product);
+            $entityManager->flush();
+
+            return new Response('Saved new product with id '.$product->getId());
+        }
+    }
+
+The following table summarizes the mapping between Doctrine metadata and
+the corresponding validation constraints:
+
++--------------------+-----------------------------------------------------------+-------------------------------------------------------------------------+
+| Doctrine attribute | Validation constraint                                     | Notes                                                                   |
++====================+===========================================================+=========================================================================+
+| ``nullable=true``  | :doc:`NotNull </reference/constraints/NotNull>`           | Relies on :doc:`the PropertyInfo component </components/property_info>` |
++--------------------+-----------------------------------------------------------+-------------------------------------------------------------------------+
+| ``type``           | :doc:`Type </reference/constraints/Type>`                 | Relies on :doc:`the PropertyInfo component </components/property_info>` |
++--------------------+-----------------------------------------------------------+-------------------------------------------------------------------------+
+| ``unique=true``    | :doc:`UniqueEntity </reference/constraints/UniqueEntity>` |                                                                         |
++--------------------+-----------------------------------------------------------+-------------------------------------------------------------------------+
+| ``length``         | :doc:`Length </reference/constraints/Length>`             |                                                                         |
++--------------------+-----------------------------------------------------------+-------------------------------------------------------------------------+
+
+Because :doc:`the Form component </forms>` as well as `API Platform`_
+internally use the Validator Component, all your forms
+and web APIs will also automatically benefit from these default constraints.
+
+.. versionadded:: 4.3
+
+    The automatic validation has been added in Symfony 4.3.
```

Comment:
what about documenting that this can be, like lot of thing in symfony, configured to reflect developer needs?

Reply:
It not yet the case (the PR in DoctrineBundle is still missing).